### PR TITLE
feat!: Add support for `node_pool_auto_config`

### PR DESCRIPTION
**Pull request by waxb**
- feat: support for `node_pool_auto_config` added
- chore: changelog updated
- chore: tests/readme updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- Add support for `node_pool_auto_config {}` with `network_tags`
+
+### Changed
+
+- BREAKING: Upgraded provider to `~> 4.34` to add support for `node_pool_auto_config`
```
Comment (mariux) [inline CHANGELOG.md]: not a breaking change as it is just a minor update...

File: versions.tf [hunk 2]
```diff
@@ -3,9 +3,16 @@
 # ----------------------------------------------------------------------------------------------------------------------
 
 terraform {
-  required_version = ">= 0.14.7, < 2.0"
+  required_version = ">= 1.0, < 2.0"
```
Comment (mariux) [inline versions.tf]:
```suggestion
  required_version = "~> 1.0"
```

File: versions.tf [hunk 3]
```diff
@@ -3,9 +3,16 @@
...
   required_providers {
-    google = "~> 4.16"
+    google = {
+      source  = "hashicorp/google"
+      version = "~> 4.34"
+    }
+    google-beta = {
+      source  = "hashicorp/google-beta"
+      version = "~> 4.34"
+    }
```
Comment (mariux) [inline versions.tf]: why do we need both providers?

File: README.tfdoc.hcl [hunk 4]
```diff
@@ -575,6 +601,38 @@ section {
           `RAPID`, `REGULAR` and `STABLE`.
         END
       }
+      variable "node_pool_auto_config" {
+        type        = any
+        default     = null
+        description = <<-END
+          (Optional) Node pool configs that apply to auto-provisioned node pools
```
Comment (mariux) [inline README.tfdoc.hcl]:
```suggestion
          Node pool configs that apply to auto-provisioned node pools
```

File: CHANGELOG.md (hunk 1 above)
Comment (waxb) [reply CHANGELOG.md]: based on a previous release in this module i thought this falls under the same category as it introduces a bigger minimum provider version requirement
> BREAKING: Upgrade provider to `~4.16` to add support for `authenticator_groups_config`

i can remove

File: versions.tf (hunk 3 above)
Comment (waxb) [reply versions.tf]: i'm still not fully familiar with our module writing tidbits yet (constantly learning and working on it! :) ) so i'm always trying to make assumptions based on other modules we have, for example this (where we use `google-beta` also):
https://github.com/mineiros-io/terraform-google-github-workload-identity/blob/main/versions.tf

File: CHANGELOG.md (hunk 1 above)
Comment (waxb) [reply CHANGELOG.md]: as mentioned, i'm trying to work on assumptions on previous work you've introduced as a team

File: main.tf [hunk 5]
```diff
@@ -208,6 +210,26 @@ resource "google_container_cluster" "cluster" {
...
+  dynamic "node_pool_auto_config" {
+    for_each = var.node_pool_auto_config != null ? [1] : []
+
+    content {
+      dynamic "network_tags" {
+        for_each = var.node_pool_auto_config.network_tags.tags != null ? [1] : []
+
+        content {
+          tags = var.node_pool_auto_config.network_tags.tags
+        }
```
Comment (mariux) [inline main.tf]: please see https://www.notion.so/Creating-new-Mineiros-Modules-e26d711cda7d43f39fe73d316eaf0d6e#d6f4820e6bd14c61bad85da5f3068cd3

```suggestion
    for_each = var.node_pool_auto_config != null ? [var.node_pool_auto_config] : []

    content {
      dynamic "network_tags" {
        for_each = try([node_pool_auto_config.value.network_tags], [])

        content {
          tags = try(network_tags.value.tags, null)
        }
```

File: README.tfdoc.hcl [hunk 6]
```diff
@@ -575,6 +601,38 @@ section {
           `RAPID`, `REGULAR` and `STABLE`.
         END
       }
+      variable "node_pool_auto_config" {
+        type        = any
```
Comment (mariux) [inline README.tfdoc.hcl]: please never reference `any` in docs... 

```suggestion
        type        = object(node_pool_auto_config)
```

File: README.tfdoc.hcl [hunk 7]
```diff
@@ -575,6 +601,38 @@ section {
...
+          node_pool_auto_config = {
+            network_tags = {
+              tags = ["foo", "bar"]
+            }
+          }
+        END
+
+        attribute "network_tags" {
+          required    = false
+          type        = any
```
Comment (mariux) [inline README.tfdoc.hcl]:
```suggestion
          type        = object(network_tags)
```

File: CHANGELOG.md (hunk 1 above)
Comment (mariux) [reply CHANGELOG.md]: it's only breaking if support for a previous major released is removed.

that's why it is a minor release in the provider ;)

Comment (waxb) [reply CHANGELOG.md]: i know, but then this wasn't a breaking change either? it was just a 4.10 -> 4.16

File: CHANGELOG.md [hunk 8]
```diff
@@ -25,7 +33,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Changed
 
-- BREAKING: Upgrade provider to `~4.16` to add support for `authenticator_groups_config`
+- Upgrade provider to `~4.16` to add support for `authenticator_groups_config`
```
Comment (mariux) [inline CHANGELOG.md]:
```suggestion
- Upgrade provider to `~> 4.16` to add support for `authenticator_groups_config`
```

**Review by zied-elouaer — DISMISSED** [review]
LGTM

File: main.tf [hunk 9]
```diff
@@ -7,6 +7,8 @@ locals {
 resource "google_container_cluster" "cluster" {
   count = var.module_enabled ? 1 : 0
 
+  provider = google-beta
+
   project = local.project
```
Comment (mariux) [inline main.tf]: please drop local project loading... the provider is falling back automatically..

File: CHANGELOG.md [hunk 10]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- Add support for `node_pool_auto_config {}` with `network_tags`
+
+### Changed
+
+- Upgraded provider to `~> 4.34` to add support for `node_pool_auto_config`
```
Comment (mariux) [inline CHANGELOG.md]: missed we really actually change to `google-beta` provider here.. not really breaking but people need to configure and pin this provider now when using the module.. so maybe BREAKING again? ;) sorry for the confusion but the message is also missing the change here ;)

File: test/terramate_google.tm.hcl [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+      description = "(Optional) The billing account to use when creating projects."
+      default     = null
+    }
+
+    terraform {
+      required_providers {
+        google = {
+          source  = "hashicorp/google"
+          version = global.provider_version_constraint
+        }
```
Comment (mariux) [inline test/terramate_google.tm.hcl:39]: we are not using google provider here but beta

**Review by mariux — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 8 above)
Comment (mariux) [inline CHANGELOG.md]:
```suggestion
- Upgrade provider to `~> 4.16` to add support for `authenticator_groups_config`
```

**Review by mariux — CHANGES_REQUESTED** [review]

**Review by zied-elouaer — DISMISSED** [review]

**Review by zied-elouaer — DISMISSED** [review]

File: CHANGELOG.md [hunk 12]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- Add support for `node_pool_auto_config {}` with `network_tags`
+
+### Changed
+
+- BREAKING: Upgraded provider to `~> 4.34` to and switched to `beta` for adding support for `node_pool_auto_config`
```
Comment (mariux) [inline CHANGELOG.md]:
```suggestion
- BREAKING: Upgraded provider to `~> 4.34` to and switched to `google-beta` Terraform Provider for adding support for `node_pool_auto_config`
```

File: test/terramate_google.tm.hcl [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+    variable "gcp_billing_account" {
+      type        = string
+      description = "(Optional) The billing account to use when creating projects."
+      default     = null
+    }
+
+    terraform {
+      required_providers {
+        google = {
+          source  = "hashicorp/google-beta"
```
Comment (mariux) [inline test/terramate_google.tm.hcl]:
```suggestion
          source  = "hashicorp/google"
```

Comment (mariux) [reply test/terramate_google.tm.hcl]: lets update from latest template again should be fine then

**Review by mariux — CHANGES_REQUESTED** [review]

**Review by soerenmartius — DISMISSED** [review]
lgtm! good job @waxb

File: README.md [hunk 14]
```diff
@@ -12,7 +12,7 @@ A [Terraform](https://www.terraform.io) module to create and manage a Google
 Kubernetes Engine (GKE) cluster with autopilot enabled.
 
 **_This module supports Terraform version 1
-and is compatible with the Terraform Google Provider version 4._**
+and is compatible with the Terraform Google Provider Beta version ~> 4.34**
```
Comment (mariux) [inline README.md:15]: we should think about how to communicate the stuff we discussed yesterday about beta and non beta setups...(just a note)

File: main.tf [hunk 15]
```diff
@@ -1,13 +1,7 @@
...
-
-locals {
-  project = var.project != null ? var.project : data.google_project.project.project_id
-}
-
 resource "google_container_cluster" "cluster" {
   count = var.module_enabled ? 1 : 0
 
-  project = local.project
+  provider = google-beta
```
Comment (mariux) [inline main.tf:4]:
```suggestion
  provider = google-beta
  
  project = var.project
```

**Review by mariux — CHANGES_REQUESTED** [review]

**Review by soerenmartius — APPROVED** [review]
lgtm

**Review by mariux — APPROVED** [review]
lgtm

**Review by zied-elouaer — APPROVED** [review]

**Comment by waxb** [comment]
thank you very much for your thorough reviews

**Pull request merged**